Swiftfin/Views/LiveTVProgramsView.swift

```diff
@@ -10,7 +10,201 @@ import Stinsen
 import SwiftUI
 
 struct LiveTVProgramsView: View {
+	@EnvironmentObject
+	var programsRouter: LiveTVProgramsCoordinator.Router
+	@StateObject
+	var viewModel = LiveTVProgramsViewModel()
+
 	var body: some View {
-		Text("Coming Soon")
+		ScrollView {
+			LazyVStack(alignment: .leading) {
+				if !viewModel.recommendedItems.isEmpty,
+				   let items = viewModel.recommendedItems
+				{
+					Text("On Now")
+						.font(.headline)
+						.fontWeight(.semibold)
+						.padding(.leading, 90)
+					ScrollView(.horizontal, showsIndicators: false) {
+						LazyHStack {
+							Spacer().frame(width: 45)
+							ForEach(items, id: \.id) { item in
+								Button {
+									if let chanId = item.channelId,
+									   let chan = viewModel.findChannel(id: chanId)
+									{
+										self.viewModel.fetchVideoPlayerViewModel(item: chan) { playerViewModel in
+											self.programsRouter.route(to: \.videoPlayer, playerViewModel)
+										}
+									}
+								} label: {
+									#if os(iOS)
+									#elseif os(tvOS)
+										LandscapeItemElement(item: item)
+									#endif
+								}
+								.buttonStyle(PlainNavigationLinkButtonStyle())
+							}
+							Spacer().frame(width: 45)
+						}
+					}.frame(height: 350)
+				}
+				if !viewModel.seriesItems.isEmpty,
+				   let items = viewModel.seriesItems
+				{
+					Text("Shows")
+						.font(.headline)
+						.fontWeight(.semibold)
+						.padding(.leading, 90)
+					ScrollView(.horizontal, showsIndicators: false) {
+						LazyHStack {
+							Spacer().frame(width: 45)
+							ForEach(items, id: \.id) { item in
+								Button {
+									if let chanId = item.channelId,
+									   let chan = viewModel.findChannel(id: chanId)
+									{
+										self.viewModel.fetchVideoPlayerViewModel(item: chan) { playerViewModel in
+											self.programsRouter.route(to: \.videoPlayer, playerViewModel)
+										}
+									}
+								} label: {
+									#if os(iOS)
+									#elseif os(tvOS)
+										LandscapeItemElement(item: item)
+									#endif
+								}
+								.buttonStyle(PlainNavigationLinkButtonStyle())
+							}
+							Spacer().frame(width: 45)
+						}
+					}.frame(height: 350)
+				}
+				if !viewModel.movieItems.isEmpty,
+				   let items = viewModel.movieItems
+				{
+					Text("Movies")
+						.font(.headline)
+						.fontWeight(.semibold)
+						.padding(.leading, 90)
+					ScrollView(.horizontal, showsIndicators: false) {
+						LazyHStack {
+							Spacer().frame(width: 45)
+							ForEach(items, id: \.id) { item in
+								Button {
+									if let chanId = item.channelId,
+									   let chan = viewModel.findChannel(id: chanId)
+									{
+										self.viewModel.fetchVideoPlayerViewModel(item: chan) { playerViewModel in
+											self.programsRouter.route(to: \.videoPlayer, playerViewModel)
+										}
+									}
+								} label: {
+									#if os(iOS)
+									#elseif os(tvOS)
```

Comment:
This file is only targeted for iOS, it will never reach tvOS, right? There are a few of these in this file.